Rename Valid language tag to Known primary language tag (#1738)
act-rules/act-rules.github.io: pages/glossary/primary-language-subtag.md

> ---

Reviewer: Git has no real notion of moving (i.e. tracking of file). It actually delete + create new file when using `git mv`. Then it assumes it is a move if the file is "close enough" to the old one.
When doing `git nv` + significant changes (as here), the changes throw git out of balance and it only sees the delete + create.

The way we've solved it previously was to have another PR that only does the `git mv` (not changing the content of the file at all, might need to update the links in the rules) first. That way there is a recorded commit in the main branch (`develop` in our case) that stays, and later some update to that file without changing the name, that git can easily track.

Not nice, but seems to be the only way to keep the history 😞 OTOH we might decide we don't really care about the history for a definition (not as important as for a rule…)

> title: Known Primary Language Tag

Reviewer: Wondering whether this should be "Known Primary Language **sub**tag" instead (so that "tag" is always the full tag, and "subtag" is the hyphen separated bits, staying closer to the RFC.

key: known-primary-language-tag
unambiguous: true
objective: true
---

> The [primary language subtag][] of a [language tag][] that exists in the [language subtag registry][] with a [Type field][] whose field-body value is `language`.

Reviewer: Given the usage in the rules, I'd go full like:
```suggestion
A language tag has a _known primary language tag_ if its [primary language subtag][] exists in the [language subtag registry][] with a [Type field][] whose field-body value is `language`.
```

> A "language tag" is here to be understood as in the first paragraph of the [RFC 5646 language tag syntax][language tag], i.e. a sequence of subtags separated by hyphens, where a subtag is any sequence of alphanumerical characters. This definition is different from the stricter [RFC 5646 syntax][language tag] (and ABNF grammar) as user agents and assistive technologies are more lenient in what they accept. This definition is consistent with the behavior of the `:lang()` pseudo-selector as defined by [Selectors Level 3][]. For example, `de-hello` would be an accepted way to indicate German in current user agents and assistive technologies, despite not being valid according to [RFC 5646 grammar][language tag]. As a consequence of this definition, however, [grandfathered tags][] are not correctly recognized as primary language subtags.

Reviewer: This needs to be tweaked a bit, notably because the discussion on the meaning of "valid" changes…
```suggestion
A "language tag" is here to be understood as in the first paragraph of the [RFC 5646 language tag syntax][language tag], i.e. a sequence of subtags separated by hyphens, where a subtag is any sequence of alphanumerical characters. [Language tag][] that are not valid according to the stricter [RFC 5646 syntax][language tag] (and ABNF grammar) definition may nonetheless have a known primary language tag. User agents and assistive technologies are more lenient in what they accept. This definition is consistent with the behavior of the `:lang()` pseudo-selector as defined by [Selectors Level 3][].

For example, `de-hello` would be an accepted way to indicate German in current user agents and assistive technologies, despite not being valid according to [RFC 5646 grammar][language tag]. It has a known primary language tag (namely, `de`).

As a consequence of this definition, however, [grandfathered tags][] do not have a known primary language tag.
```
